### PR TITLE
Add Jira Service Management connector

### PR DESCRIPTION
**Pull request by aneeshafedo**
# Description

Add Jira Service Management Connectors

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/232

One line release note: 
- Add Jira Service Management connector

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

**Test Configuration**:
* Ballerina Version: Ballerina Swan Lake Beta 6
* Operating System: Linux
* Java SDK: Java 11

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/jira.servicemanagement/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Support//Customer Support", "Cost/Freemium"]
+org = "ballerinax"
+name = "jira.servicemanagement"
+distribution = "slbeta6"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "0.1.0"
```
Comment (RolandHewage) [inline openapi/jira.servicemanagement/Ballerina.toml:8]: Shall we update the version to 1.0.0 & build again

File: openapi/jira.servicemanagement/client.bal [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+    final http:Client clientEp;
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. 
+    # Create an [Atlassian Account](https://id.atlassian.com/signup) 
+    # and obtain tokens by following [this guide](https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/).
+    #
+    # + clientConfig - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ClientConfig clientConfig, string serviceUrl = "https://your-domain.atlassian.net") returns error? {
```
Comment (RolandHewage) [inline openapi/jira.servicemanagement/client.bal:66]: Shall we remove the default value for the service URL since this can vary based on the user domain.

**Review by RolandHewage — COMMENTED** [review]
Few changes requested

File: openapi/jira.servicemanagement/client.bal [hunk 3]
```diff
@@ -0,0 +1,827 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sachinira) [inline openapi/jira.servicemanagement/client.bal:1]: Change the year to 2022. Check all possible places

File: openapi/jira.servicemanagement/Ballerina.toml (hunk 1 above)
Comment (aneeshafedo) [reply openapi/jira.servicemanagement/Ballerina.toml]: Resolved

File: openapi/jira.servicemanagement/client.bal (hunk 2 above)
Comment (aneeshafedo) [reply openapi/jira.servicemanagement/client.bal]: Resolved. Thank you

File: openapi/jira.servicemanagement/client.bal (hunk 3 above)
Comment (aneeshafedo) [reply openapi/jira.servicemanagement/client.bal:1]: Updated. Thank you

**Review by abeykoon — APPROVED** [review]

**Pull request merged**